pkg/auth/metadata.go

```diff
@@ -46,108 +43,83 @@ func generateInstallationID(installationID string) string {
 	return installationID
 }
 
-func insertOrGetInstallationID(tx db.Tx, installationID string) (string, error) {
-	res, err := tx.Exec(`INSERT INTO auth_installation_metadata (key_name, key_value)
-			VALUES ($1,$2)
-			ON CONFLICT DO NOTHING`,
-		InstallationIDKeyName, installationID)
+func (m *KVMetadataManager) insertOrGetInstallationID(ctx context.Context, installationID string) (string, error) {
+	err := m.store.SetIf(ctx, []byte(model.PartitionKey), []byte(installationID), []byte(installationID), nil)
```

Comment:
Maybe we should define a proto model for this, so we can support future changes as needed?